cloud/passportauth/src/user/BaseUser.ts

```diff
@@ -0,0 +1,32 @@
+export interface BaseUser {
+  /**
+   * Retrieve's the user's id.
+   *
+   * @param loginId {string} - User's login id (i.e. username, email, etc)
+   * @returns {Promise<string>} - A unique id used to identify the user
+   */
+  getId(loginId: string): Promise<string|null>;
+
+  /**
+   * Retrieves the user's login id
+   *
+   * @returns {Promise<string>} - A unique id used by the user for login
+   */
+  getLoginId(): Promise<string|null>;
+
+  /**
+   * Retrieves the user's hashed password
+   *
+   * @returns {Promise<string>} - User's hashed password
+   */
+  getPasswordHash(): Promise<string|null>;
```

Comment:
should be stateless; you need to pass any user to interface
getPasswordHash(user) {return user.password}